src/particles/elements/Aperture.H

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_APERTURE_H
+#define IMPACTX_APERTURE_H
+
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thin.H"
+#include "mixin/nofinalize.H"
+
+#include <ablastr/particles/IndexHandling.H>
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+namespace impactx
+{
+    struct Aperture
+    : public elements::BeamOptic<Aperture>,
+      public elements::Thin,
+      public elements::NoFinalize
+    {
+        static constexpr auto name = "Aperture";
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** A thin collimator element that applies a transverse aperture boundary.
+         *  Particles outside the boundary are considered lost.
+         *
+         * @param shape aperture shape (0 rectangular, 1 elliptical)
+         * @param xmax maximum value of horizontal coordinate (m)
+         * @param ymax maximum value of vertical coordinate (m)
+         */
+        Aperture( int const shape,
+                   amrex::ParticleReal const xmax,
+                   amrex::ParticleReal const ymax )
```

Comment:
We should move `shape` last here, too.

I would suggest we can make it an enum class or string instead of an int, because then the option is self-describing at its call sites.

Reply:
done :)